src/components/Gallery/Image.tsx

```diff
@@ -1,30 +1,42 @@
 import React, { useState } from 'react';
 import { sanitizeUrl } from '@braintree/sanitize-url';
 
-import { ModalComponent as ModalWrapper } from './ModalWrapper';
-
-export type ImageProps = {
-  /** The text fallback for the image */
-  fallback?: string;
-  /** The full size image url */
-  image_url?: string;
-  /** The thumb url */
-  thumb_url?: string;
-};
+import { Modal } from '../Modal';
+import { ModalGallery as DefaultModalGallery } from './ModalGallery';
+import { useComponentContext } from '../../context';
+
+import type { Attachment } from 'stream-chat';
+import type { DefaultStreamChatGenerics } from '../../types/types';
+
+export type ImageProps<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics
+> =
+  | {
+      /** The text fallback for the image */
+      fallback?: string;
+      /** The full size image url */
+      image_url?: string;
+      /** The thumb url */
+      thumb_url?: string;
+    }
+  | Attachment<StreamChatGenerics>;
 
 /**
  * A simple component that displays an image.
  */
-export const ImageComponent = (props: ImageProps) => {
+export const ImageComponent = <
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics
+>(
+  props: ImageProps<StreamChatGenerics>,
+) => {
+  const { fallback, image_url, thumb_url } = props;
```

Comment:
@arnautov-anton I would have to then create a new object to be passed to the `ModalGallery` component few lines below:

so instead of this:
```
<Modal onClose={toggleModal} open={modalIsOpen}>
        <ModalGallery images={[props]} index={0} />
      </Modal>
```

I would have to do this (as props obj is not available):

```
<Modal onClose={toggleModal} open={modalIsOpen}>
        <ModalGallery images={[{ fallback, image_url, thumb_url }]} index={0} />
      </Modal>
```

I would rather stick with the current state for better readability.